lib/ansible/modules/system/firewalld.py

```diff
@@ -799,15 +879,26 @@ def main():
             immediate=dict(type='bool', default=False),
             source=dict(required=False, default=None),
             permanent=dict(type='bool', required=False, default=None),
-            state=dict(choices=['enabled', 'disabled', 'present', 'absent'], required=True),
+            state=dict(choices=['enabled', 'disabled'], required=True),
             timeout=dict(type='int', required=False, default=0),
             interface=dict(required=False, default=None),
             masquerade=dict(required=False, default=None),
             offline=dict(type='bool', required=False, default=None),
+            chain=dict(required=False, default=None),
+            table=dict(required=False, default=None),
+            direct_rule=dict(required=False, default=None),
+            fw_family=dict(required=False, default=None),
+            rule_priority=dict(type='int', required=False, default=0)
         ),
         supports_check_mode=True
     )
 
+    if import_failure:
```

Comment:
This should not be moved, the order of the conditionals are on purpose because the prevent from different error scenarios based on remote system state.